napari/_vispy/layers/labels.py

```diff
@@ -68,8 +73,12 @@
     // we get a different value:
     // - if it's the empty key, exit;
     // - otherwise, it's a hash collision: continue searching
-    while ((abs(found - t) > 1e-8) && (abs(found - empty) > 1e-8)) {
+    float initial_t = t;
+    while ((abs(found - initial_t) > 1e-8) && (abs(found - empty) > 1e-8)) {
         t = t + 1;
+        if (t - initial_t > LUT_shape.x * LUT_shape.y) {
+            return vec4(0);
+        }
```

Comment:
```suggestion
    float initial_t = t;
    float lut_size = LUT_shape.x * LUT_shape.y
    while ((abs(found - initial_t) > 1e-8) && (abs(found - empty) > 1e-8)) {
        t = t + 1;
        if (t - initial_t > lut_size) {
            return vec4(0);
        }
```

maybe a micro-optimisation but since that condition is there for *many* loop locations, I thought it might be worth doing?

Reply:
I'm pretty sure this is easily optimized by the compiler, but I guess it doesn't hurt...